volgen/volgen.go

```diff
@@ -1,9 +1,30 @@
 package volgen
 
 import (
+	"fmt"
 	"io"
 )
 
 func (x Xlator) DumpGraph(w io.Writer) {
-	_, _ = w.Write([]byte("Dump function not implemented yet"))
+
+	for _, xl := range x.Children {
+		xl.DumpGraph(w)
+	}
+
+	fmt.Fprintf(w, "volume %s\n", x.Name)
+	fmt.Fprintf(w, "   type %s", x.Type)
+
+	for key, value := range x.Options {
+		fmt.Fprintf(w, "\n   option %s %s", key, value)
+	}
+
+	if x.Children != nil {
+		fmt.Fprintf(w, "\n   subvolumes")
+
+		for i := len(x.Children) - 1; i >= 0; i-- {
```

Comment:
Please keep the code as simple as possible. There is no need to do a reverse traversal here IMO.